### PR TITLE
add X-Lakefs-Client header to metaclient requests

### LINKED ISSUE #4524: Add X-Lakefs-Client header to requests by lakeFS client
Add the header to requests sent to the lakeFS server by the following clients:
1. HadoopFS
2. lakeFS metadata client - GC
3. lakeFS metadata client - Export
4. lakeFS metadata client - other

### PR DESCRIPTION
**Pull request by johnnyaug**
Tested from Databricks on a locally-built jar.

Resolves #4524.

### CODE REVIEW

**Review by arielshaqed — APPROVED** [review]
Neat, but can we be more consistent in how we name our clients please?

File: clients/spark/core/src/main/scala/io/treeverse/clients/LakeFSInputFormat.scala [hunk 1]
```diff
@@ -157,7 +157,8 @@ class LakeFSCommitInputFormat extends LakeFSBaseInputFormat {
         conf.get(LAKEFS_CONF_API_ACCESS_KEY_KEY),
         conf.get(LAKEFS_CONF_API_SECRET_KEY_KEY),
         conf.get(LAKEFS_CONF_API_CONNECTION_TIMEOUT_SEC_KEY),
-        conf.get(LAKEFS_CONF_API_READ_TIMEOUT_SEC_KEY)
+        conf.get(LAKEFS_CONF_API_READ_TIMEOUT_SEC_KEY),
+        "input_format"
```
[the remark this answers is not in the document]
Comment (johnnyaug) [reply clients/spark/core/src/main/scala/io/treeverse/clients/LakeFSInputFormat.scala]: Refactored. Careful what you wish for :)

File: clients/spark/examples/src/main/scala/io/treeverse/clients/examples/Export.scala [hunk 2]
```diff
@@ -45,7 +45,13 @@ object Export extends App {
...
 
     val apiClient = ApiClient.get(
-      APIConfigurations(endpoint, accessKey, secretKey, connectionTimeoutSec, readTimeoutSec) /**/
+      APIConfigurations(endpoint,
+                        accessKey,
+                        secretKey,
+                        connectionTimeoutSec,
+                        readTimeoutSec,
+                        "exporter"
+                       ) /**/
```
[the remark this answers is not in the document]
Comment (johnnyaug) [reply clients/spark/examples/src/main/scala/io/treeverse/clients/examples/Export.scala]: Done

File: clients/spark/build.sbt [hunk 3]
```diff
@@ -22,10 +22,13 @@ def settingsToCompileIn(dir: String, flavour: String = "") = {
     Compile / PB.includePaths += (Compile / resourceDirectory).value,
     Compile / PB.protoSources += (Compile / resourceDirectory).value
   )
-  lazy val flavourSettings = if (flavour != "")
-    Seq(Compile / unmanagedSourceDirectories += (ThisBuild / baseDirectory).value /dir / "src" / "main" / flavour / "scala")
-  else
-    Seq()
+  lazy val flavourSettings =
```
[the remark this answers is not in the document]
Comment (johnnyaug) [reply clients/spark/build.sbt]: reverted

File: clients/spark/core/src/main/scala/io/treeverse/clients/GarbageCollector.scala [hunk 4]
```diff
@@ -269,7 +269,8 @@ object GarbageCollector {
           .toEpochSecond(ZoneOffset.UTC)
       )
     }
-    val apiConf = APIConfigurations(apiURL, accessKey, secretKey, connectionTimeout, readTimeout)
+    val apiConf =
+      APIConfigurations(apiURL, accessKey, secretKey, connectionTimeout, readTimeout, "gc")
```
Comment (arielshaqed) [inline clients/spark/core/src/main/scala/io/treeverse/clients/GarbageCollector.scala:273]: Let's use an APP_NAME constant here too please, to make it less confusing.

**Review by arielshaqed — APPROVED** [review]
Thanks!

(I understand _why_ you added the refactoring, but I'd rather have reviewed two PRs, one refactoring the params and the next adding the app name.)

**Pull request merged**